### PR TITLE
Change default build to *not* include Rust API.

### PR DESCRIPTION
**Pull request by NlightNFotis**
In offline discussions, concerns were raised about Rust/Cargo being a build-time dependency of CBMC, and that users should have the (default) option of doing a CBMC build without Rust installed in their systems.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]

File: COMPILING.md [hunk 1]
```diff
@@ -562,3 +562,29 @@ then run make as follows.
...
 make test
 ```
+
+## Rust API
+
+CBMC is now offering a [Rust API](src/libcprover-rust/). To build that along with
+CBMC, you need two things:
+
+- Rust/Cargo, instructions the installation of which can be found [here](https://www.rust-lang.org/tools/install), and
+- CMake (the Rust API doesn't support being built with `Make` yet)
```
Comment (tautschnig) [inline COMPILING.md:572]: Shouldn't this also specify a version for CMake (unless 3.19 is available on all reasonable systems today)?

File: COMPILING.md [hunk 2]
```diff
@@ -562,3 +562,29 @@ then run make as follows.
...
+-- Found Rust: /opt/homebrew/bin/rustc (found version "1.66.1") 
+-- Using Corrosion as a subdirectory
+-- Configuring done
+-- Generating done
+-- Build files have been written to: cbmc/build
+```
+
+If you come across any issues during the configuration or the build of the Rust API,
+please let us know by filing a [bug report](https://github.com/diffblue/cbmc/issues/new)
+and mark it with the label `Rust API` if possible.
```
Comment (tautschnig) [inline COMPILING.md:592]: I'm not sure non-maintainers have the ability to add labels?

Comment (NlightNFotis) [reply COMPILING.md:592]: I'm not sure either, hence why the `if possible`, but I also know that other developers/maintainers refer to this document on some occasions, so the label information is useful enough to remain IMO.

File: COMPILING.md (hunk 1 above)
Comment (NlightNFotis) [reply COMPILING.md:572]: I will change the documentation and issue a `CMake` error in this case.

**Pull request merged**